client/src/components/ToolRecommendation.vue

```diff
@@ -86,32 +86,60 @@ export default {
             const duration = 750;
             const maxTextLength = 20;
             const svg = d3.select("#tool-recommendation").append("svg").attr("class", "tree-size").append("g");
-            const gElem = svg[0][0];
+            const gElem = svg._groups[0][0];
```

Comment:
Properties prefixed with `_` usually indicate internal properties, which are undocumented, and may change even on minor releases. A quick search through the d3 docs seems to confirm this is also the case here.

There should be an official way to do this, which would be preferable. Could you link to where you found this property?

Reply:
Yes, it also looks weird to me. Currently trying to find its alternative. I found it while looking inside the `svg` object.

Reply:
Using `d3.select(...).node()` offers an easier but effective solution.